controllers/wso2is_controller.go

```diff
@@ -0,0 +1,592 @@
+			AccessModes: []corev1.PersistentVolumeAccessMode{
+				"ReadWriteMany",
+			},
+			Resources: corev1.ResourceRequirements{
+				Requests: corev1.ResourceList{
+					corev1.ResourceStorage: resource.MustParse("1Gi"),
+				},
+			},
+		},
+	}
+	ctrl.SetControllerReference(&m, pvc, r.Scheme)
+	return pvc
+}
+
+// addConfigMap adds a new ConfigMap
+func (r *Wso2IsReconciler) addConfigMap(m wso2v1beta1.Wso2Is, logger logr.Logger) *corev1.ConfigMap {
+	configMap := &corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "identity-server-conf",
+			Namespace: m.Namespace,
+		},
+		Data: map[string]string{
+			"deployment.toml": getTomlConfig(m.Spec, logger),
+		},
+	}
+	ctrl.SetControllerReference(&m, configMap, r.Scheme)
+	return configMap
+}
+
+func getTomlConfig(spec wso2v1beta1.Wso2IsSpec, logger logr.Logger) string {
+	if len(spec.TomlConfig) == 0 {
+		buf := new(bytes.Buffer)
+		if err := toml.NewEncoder(buf).Encode(spec.Configurations); err != nil {
+			log.Fatal(err)
+		}
+		logger.Info(buf.String())
+		return buf.String()
+	} else {
+		return spec.TomlConfig
+	}
+}
+
+// addNewIngress adds a new Ingress Controller
+func (r *Wso2IsReconciler) addNewIngress(m wso2v1beta1.Wso2Is) *v1beta1.Ingress {
+	ingress := &v1beta1.Ingress{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "wso2is-ingress",
+			Namespace: m.Namespace,
+			Annotations: map[string]string{
+				"kubernetes.io/ingress.class":                     "nginx",
+				"nginx.ingress.kubernetes.io/backend-protocol":    "HTTPS",
+				"nginx.ingress.kubernetes.io/affinity":            "cookie",
+				"nginx.ingress.kubernetes.io/session-cookie-name": "route",
+				"nginx.ingress.kubernetes.io/session-cookie-hash": "sha1",
+			},
+		},
+		Spec: v1beta1.IngressSpec{
+			TLS: []v1beta1.IngressTLS{
+				{
+					Hosts: []string{m.Spec.Configurations.Host},
+				},
+			},
+			Rules: []v1beta1.IngressRule{
+				{
+					Host: m.Spec.Configurations.Host,
+					IngressRuleValue: v1beta1.IngressRuleValue{
+						HTTP: &v1beta1.HTTPIngressRuleValue{
+							Paths: []v1beta1.HTTPIngressPath{{
+								Path: "/",
+								Backend: v1beta1.IngressBackend{
+									ServiceName: "wso2is-service",
+									ServicePort: intstr.IntOrString{
+										IntVal: 9443,
+									},
+								},
+							}},
+						},
+					},
+				},
+			},
+		},
+	}
+	ctrl.SetControllerReference(&m, ingress, r.Scheme)
+	return ingress
+}
+
+// addNewService adds a new Service
+func (r *Wso2IsReconciler) addNewService(m wso2v1beta1.Wso2Is) *corev1.Service {
+	svc := &corev1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "wso2is-service",
+			Namespace: m.Namespace,
+		},
+		Spec: corev1.ServiceSpec{
+			Ports: []corev1.ServicePort{{
+				Name:     "servlet-http",
+				Protocol: "TCP",
+				Port:     9763,
+				TargetPort: intstr.IntOrString{
+					IntVal: 9763,
+				},
+			}, {
+				Name:     "servlet-https",
+				Protocol: "TCP",
+				Port:     9443,
+				TargetPort: intstr.IntOrString{
+					IntVal: 9443,
```

Comment:
Shall we move these port values to constants?

Reply:
Done